rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -318,6 +321,18 @@
         }
       });
 
+      // We can't access these variables from within the tooltip mouse handlers
+      // because the handler is only set on bar creation, not bar update, so the closure will be out of date.
+      // Instead the count will be added to the histogram as an extra property so it's available as an argument.
+      var totalResponseCount = _.reduce(histogram, function(total, item) {
+        return total + item.count
+      }, 0);
+      histogram = histogram.map(function(item) {
+        return _.assign({}, item, {
+          totalResponseCount: totalResponseCount
+        });
+      });
```

Comment:
Shouldn't we just calculate the total # of responses and the percentages in the API?

Reply:
Yeah that makes more sense

Reply:
Actually I don't think this would improve things much. The `totalResponseCount` assignment there is because the local variables in this function aren't accessible from within the callback, including `selectedRun`. If the calculation was done in the backend it would need to put a dict in the REST API and the `var totalResponseCount` line would become `var totalResponseCount = state.total_counts[selectedRun] || 0`

Reply:
Sure, I made the server side change